packages/nvidia_nat_autogen/tests/test_llm_autogen.py

```
@@ -0,0 +1,637 @@
+from nat.data_models.thinking_mixin import ThinkingMixin
+from nat.llm.azure_openai_llm import AzureOpenAIModelConfig
+from nat.llm.nim_llm import NIMModelConfig
+from nat.llm.openai_llm import OpenAIModelConfig
+from nat.plugins.autogen.llm import _patch_autogen_client_based_on_config
+
+
+class MockRetryConfig(LLMBaseConfig, RetryMixin):
+    """Mock config with retry mixin."""
+
+    num_retries: int = 3
+    retry_on_status_codes: list = [500, 502, 503]
```

Ruff (0.14.8)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)